iceoryx_hoofs/test/moduletests/test_cxx_expected.cpp

```diff
@@ -334,22 +355,50 @@ TEST_F(expected_test, ErrorTypeOnlyCreateErrorLeadsToError)
     ASSERT_THAT(sut.get_error(), Eq(TestError::ERROR2));
 }
 
-TEST_F(expected_test, ErrorTypeOnlyMoveCtorLeadsToInvalidState)
+TEST_F(expected_test, ErrorTypeOnlyCreateValueWithoutValueMoveCtorLeadsToNoError)
 {
-    auto sut = expected<TestError>::create_error(TestError::ERROR2);
-    auto movedValue{std::move(sut)};
-    IOX_DISCARD_RESULT(movedValue);
-    ASSERT_TRUE(sut.has_error());
-    EXPECT_THAT(sut.get_error(), Eq(TestError::INVALID_STATE));
+    auto sutSource = expected<NonTrivialTestClass>::create_value();
+    auto sutDestination{std::move(sutSource)};
+    EXPECT_FALSE(sutSource.has_error());
+    EXPECT_FALSE(sutDestination.has_error());
+}
+
+TEST_F(expected_test, ErrorTypeOnlyCreateValueWithoutValueMoveAssignmentLeadsToNoError)
+{
+    auto sutSource = expected<NonTrivialTestClass>::create_value();
+    auto sutDestination = std::move(sutSource);
+    EXPECT_FALSE(sutSource.has_error());
+    EXPECT_FALSE(sutDestination.has_error());
+}
+
+TEST_F(expected_test, ErrorTypeOnlyMoveCtorLeadsMovedSource)
+{
+    constexpr int A{111};
+    constexpr int B{112};
+    auto sutSource = expected<NonTrivialTestClass>::create_error(A, B);
+    auto sutDestination{std::move(sutSource)};
+
+    ASSERT_TRUE(sutSource.has_error());
+    EXPECT_TRUE(sutSource.get_error().m_moved);
+    ASSERT_TRUE(sutDestination.has_error());
+    EXPECT_FALSE(sutDestination.get_error().m_moved);
+    EXPECT_EQ(sutDestination.get_error().m_a, A);
+    EXPECT_EQ(sutDestination.get_error().m_b, B);
 }
 
 TEST_F(expected_test, ErrorTypeOnlyMoveAssignmentLeadsToInvalidState)
 {
-    auto sut = expected<TestError>::create_error(TestError::ERROR1);
-    auto movedValue = std::move(sut);
-    IOX_DISCARD_RESULT(movedValue);
-    ASSERT_TRUE(sut.has_error());
-    EXPECT_THAT(sut.get_error(), Eq(TestError::INVALID_STATE));
+    constexpr int A{222};
+    constexpr int B{223};
+    auto sutSource = expected<NonTrivialTestClass>::create_error(A, B);
+    auto sutDestination = std::move(sutSource);
+
+    ASSERT_TRUE(sutSource.has_error());
+    EXPECT_TRUE(sutSource.get_error().m_moved);
+    ASSERT_TRUE(sutDestination.has_error());
+    EXPECT_FALSE(sutDestination.get_error().m_moved);
+    EXPECT_EQ(sutDestination.get_error().m_a, A);
+    EXPECT_EQ(sutDestination.get_error().m_b, B);
```

Comment:
Could you please add a unit test cases which uses an `expected<EnumClassType>::create_error(..)`?